internal/manifests/collector/horizontalpodautoscaler.go

```diff
@@ -93,19 +95,24 @@ func HorizontalPodAutoscaler(params manifests.Params) client.Object {
 			ScaleTargetRef: autoscalingv2.CrossVersionObjectReference{
 				APIVersion: v1alpha1.GroupVersion.String(),
 				Kind:       "OpenTelemetryCollector",
-				Name:       naming.OpenTelemetryCollector(params.OtelCol.Name),
+				Name:       naming.OpenTelemetryCollector(otelCol.Name),
 			},
-			MinReplicas: params.OtelCol.Spec.Autoscaler.MinReplicas,
-			MaxReplicas: *params.OtelCol.Spec.Autoscaler.MaxReplicas,
-			Metrics:     metrics,
+			MinReplicas: otelCol.Spec.OpenTelemetryCommonFields.Autoscaler.MinReplicas,
+			MaxReplicas: func(max *int32) int32 {
```

Comment:
can we make this a separate function? Why did we not need to do this before?

Reply:
The field `MaxReplicas` is in `v1alpha2` its now  a value. The field  `k8s.io/api/autoscaler/v2.HorizontalPodAutoscalerSpec.MaxReplicas` is a value too.

https://github.com/open-telemetry/opentelemetry-operator/blob/d6f3c31a072979783461dd9d3b23d3e12aeb1f26/apis/v1alpha1/opentelemetrycollector_types.go#L473-L479

https://github.com/open-telemetry/opentelemetry-operator/blob/d6f3c31a072979783461dd9d3b23d3e12aeb1f26/apis/v1alpha2/common.go#L47-L53

Reply:
Agree about moving to a different function

Reply:
It's just to make it more readable

Reply:
From my point of view this is quite a common thing.